Core/src/org/sleuthkit/autopsy/modules/vmextractor/VMExtractorIngestModule.java

```diff
@@ -237,6 +239,46 @@ public ProcessResult process(Content dataSource, DataSourceIngestModuleProgress
         }
         return vmFiles;
     }
+    
+    /**
+     * Check all the files and if a file is a vhd then check to make sure it is a valid vhd using the mimetype. We are not
+     * checking the mimetype for VMDK's at this point in time.
+     * 
+     * @param vmFiles List of virtual machine abstract files to look at
+     * 
+     * @return List of abstract files of virtual machine files.
+     */
+    private static List<AbstractFile> removeNonVMFiles(List<AbstractFile> vmFiles) {
+        List<AbstractFile> vFile = new ArrayList<>();
+
+        for (AbstractFile vmFile : vmFiles) {
+            if (vmFile.getNameExtension().equalsIgnoreCase("vhd")) {
+                String fileMimeType = vmFile.getMIMEType();
+                if (fileMimeType == null) {
+                    FileTypeDetector fileTypeDetector = null;
+                    try {
+                        fileTypeDetector = new FileTypeDetector();
+                    } catch (FileTypeDetector.FileTypeDetectorInitException ex) {
+                        logger.log(Level.WARNING, String.format("Unable to create file type detector for determining MIME type for file %s with id of %d", vmFile.getName(), vmFile.getId()));
```

Comment:
If this error occurs we will get an NPE on the next line.

Reply:
Good catch, added abstract file to list and then continue on with loop to avoid the NPE.